src/app/containers/Figure/index.jsx

```diff
@@ -15,10 +15,10 @@ const renderCaption = captionValue =>
     <Text text={captionValue} paragraphOverride={Caption} />
   ) : null;
 
-const FigureContainer = ({ src, alt, ratio, copyright, caption }) => (
+const FigureContainer = ({ src, srcSet, alt, ratio, copyright, caption }) => (
   <Figure>
     <ImagePlaceholder ratio={ratio}>
-      <Image alt={alt} src={src} />
```

Comment:
Been thinking a bit about this. `<picture>` / `srcset` are great but they don't align with our business requirements.

| Scenario | `srcset` | Bespoke JS |
|----------|----------|-------------|
| User on modern browser | Optimised image delivered immediately (lazyload only possible with JS override) | Lazyload optimised image delivered |
| User on modern browser, JS disabled | Optimised image delivered immediately (probably what we want) | `<noscript>` fallback |
| User on old desktop browser | Non-optimal - too small - (`src`) image delivered immediately (lazyload only possible with JS override) | Lazyload optimised image delivered |
| User on old mobile browser | Non-optimal - too big - (`src`) image delivered immediately (lazyload only possible with JS override) | Lazyload optimised image delivered |

Because IE11 and Opera Mini are such a large proportion of our audience (the latter particularly on the WS sites), and [they don't support `srcset`](https://caniuse.com/#search=srcset), markup alone is not a practical solution - not to mention the spec doesn't currently support lazy-loading out of the box (`src`/`srcset` will download the image url immediately).

I think a bespoke solution is probably the way forward at this point. [Something like this](https://developers.google.com/web/fundamentals/performance/lazy-loading-guidance/images-and-video/):

```html
<head>
  <style>
    .lazy-load { display: none }
    .js .lazy-load { display: block }
  </style>
  <script>document.body.className += 'js'</script>
</head>

...

<main>
  <img class="lazy-load" src="placeholder.png" data-srcset="img1.jpg, img2.jpg" />
  <noscript><img src="fallback.jpg" srcset="img1.jpg, img2.jpg" /></noscript>
</main>
...

<footer>
  <script> // lazy-load logic goes here </script>
</footer>
```

This _should_ fulfil my assumptions of what the "ideal product" is, bearing in mind technical implementations:

1. Users with modern browsers should get placeholder, then lazy-loaded image± at a size optimised for their viewport. (i.e. load quickly and don't waste data)
2. Users with modern browsers, who disable JS, should immediately load image at a size optimised for their viewport. (i.e. get the image, not be stuck with a placeholder image)
3. Users with older desktop browsers (IE11), should get the full experience (as per 1)
4. Users with older desktop browsers (IE11), who disable JS, should immediately load a one-size fallback image. (i.e. get an image - even if it's a bit too small - and not be stuck with a placeholder image)
5. Users with older mobile browsers (Opera Mini), should get the full experience (as per 1)
6. Users with older mobile browsers (Opera Mini), who disable JS, should immediately load a one-size fallback image (i.e. get an image - even if it's a bit too _big_ - and not be stuck with a placeholder image)

± lazy-loaded implementation TBC (e.g. when 1 full scroll away from being in view).

I know this investigation is aimed at a responsive images solution, and not necessarily lazy-loading, but I think solutions for one won't necessarily gel with the other and we have to keep both in mind. The actual business logic of the lazy-loading itself can be tackled later though.

Reply:
NB: We should also be considering "User on old mobile browser without JS", but that falls into the `<noscript> src` value which is optimal. 

On point 5, Opera mini runs all JS on the [Opera Mini server](https://dev.opera.com/articles/opera-mini-and-javascript/#figure-1) and only give the user the payload so without investigation/experimentation I'm not convinced they would get the lazyload functionality and may end up with the fallback or only the placeholder. 

>Much of the reason for Mini’s popularity is that it is capable of rendering complex web pages, even on low-spec devices. It does this by using a proxy-based architecture (see Figure 1). Requests from the user’s handset pass through the carrier’s internet gateway on their way to Opera’s transcoding servers. These servers then forward the request to the server. - [Dev.Opera](https://dev.opera.com/articles/opera-mini-and-javascript/#figure-1)

It's also important to note our current placeholder logic* uses a <div> not an actual placeholder <img>.

* We create a wrapping <div> which matches the height of the image and then adds a base64 SVG as the background image of the <div>. This way there is a grey placeholder in all instances and in browser that support SVG there are BBC blocks - See Figure 1

I think we can be more concise (ignoring mobile and desktop differences) with the Scenarios by defining them in the following way:

| Scenario                                    | Description                                                                       |
| ------------------------------------------- | --------------------------------------------------------------------------------- |
| Supports srcset and runs                    | Gets full experience of a placeholder, lazyload and responsive images from srcset |
| Supports srcset and does not run JS         | Gets a placeholder and responsive image from <noscript> srcset                                    |
| Does not support srcset and runs JS         | Gets a placeholder, lazyload and single width image from src                      |
| Does not support srcset and does not run JS | Gets placeholder and  a single width image from <noscript> src                                     |

## What the data says (oct 2018) - WIP these will update shortly
The percentage of browsers per service that don't support the <img> attribute `srcset`

| Service                     | Percentage |
| --------------------------- | ---------- |
| korean              | 12.500     |
| news              | 7.287     |
| news-cymru              | 9.492     |
| news-gnl-japanese              | 12.347     |
| news-naidheachdan              | 8.726     |
| news-ws-afaanoromoo              | 25.730     |
| news-ws-afrique              | 29.398     |
| news-ws-amharic              | 17.195     |
| news-ws-arabic              | 4.879     |
| news-ws-azeri              | 3.838     |
| news-ws-bengali              | 9.714     |
| news-ws-burmese              | 3.870     |
| news-ws-gahuza              | 34.130     |
| news-ws-gujarati              | 0.331     |
| news-ws-hausa              | 80.336     |
| news-ws-hindi              | 3.286     |
| news-ws-igbo              | 8.387     |
| news-ws-indonesia              | 5.716     |
| news-ws-korean              | 6.261     |
| news-ws-kyrgyz              | 3.246     |
| news-ws-marathi              | 1.430     |
| news-ws-mundo              | 1.964     |
| news-ws-nepali              | 2.616     |
| news-ws-pashto              | 5.705     |
| news-ws-persian              | 4.302     |
| news-ws-pidgin              | 6.663     |
| news-ws-portuguese              | 1.832     |
| news-ws-punjabi              | 0.737     |
| news-ws-russian              | 3.307     |
| news-ws-serbian              | 1.441     |
| news-ws-sinhala              | 3.015     |
| news-ws-somali              | 8.112     |
| news-ws-swahili              | 37.672     |
| news-ws-tamil              | 2.825     |
| news-ws-telugu              | 0.692     |
| news-ws-tigrinya              | 6.904     |
| news-ws-turkce              | 2.482     |
| news-ws-ukchina              | 2.542     |
| news-ws-ukrainian              | 2.645     |
| news-ws-urdu              | 5.134     |
| news-ws-uzbek              | 16.990     |
| news-ws-vietnamese              | 2.919     |
| news-ws-yoruba              | 10.870     |
| news-ws-zhongwen              | 3.467     |

## Figure 1 
![image](https://user-images.githubusercontent.com/7791726/48481555-eb99b600-e805-11e8-9349-caa0ea57af4c.png)

Reply:
Following up the above simplified scenarios I think the following would cover all use cases:
```
  <div class="placeholder">
    <img class="lazy-load" src="placeholder.png" data-srcset="img1.jpg, img2.jpg" />
    <noscript><img src="default-image.jpg" srcset="img1.jpg, img2.jpg" /></noscript>
  </div>
```